### PR TITLE
Use pseudocode to document `transmute`, `transmute_mut`

### PR DESCRIPTION
**Pull request by jswrenn**
Partially addresses #1046.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -4347,14 +4347,31 @@ mod simd {
...
-/// `FromBytes`.
+/// This macro behaves like an invocation of this function:
+///
+/// ```ignore
+/// const fn transmute<Src, Dst>(src: Src) -> Dst
+/// where
+///     Src: IntoBytes,
+///     Dst: FromBytes,
+///     size_of::<Src>() == size_of::<Dst>(),
+///     align_of::<Src>() >= align_of::<Dst>(),
```
Comment (joshlf) [inline src/lib.rs:4358]:
```suggestion
```

File: src/lib.rs [hunk 2]
```diff
@@ -4562,13 +4579,27 @@ macro_rules! transmute_ref {
...
+/// This macro behaves like an invocation of this function:
 ///
-/// The lifetime of the input type, `&mut T`, must be the same as or outlive the
-/// lifetime of the output type, `&mut U`.
+/// ```ignore
+/// const fn transmute_mut<'src, 'dst, Src, Dst>(src: &'src mut Src) -> &'dst mut Dst
+/// where
+///     'src: 'dst,
+///     Src: IntoBytes + FromBytes + NoCell,
+///     Dst: FromBytes + IntoBytes + NoCell,
```
Comment (joshlf) [inline src/lib.rs:4589]:
```suggestion
///     Src: FromBytes + IntoBytes + NoCell,
///     Dst: FromBytes + IntoBytes + NoCell,
```

IMO it's visually clearer this way that both types must implement the same set of traits.

**Review by joshlf — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (jswrenn) [reply src/lib.rs]: Done

File: src/lib.rs (hunk 2 above)
Comment (jswrenn) [reply src/lib.rs]: Done

File: src/lib.rs [hunk 3]
```diff
@@ -4562,13 +4578,27 @@ macro_rules! transmute_ref {
...
+/// This macro behaves like an invocation of this function:
 ///
-/// The lifetime of the input type, `&mut T`, must be the same as or outlive the
-/// lifetime of the output type, `&mut U`.
+/// ```ignore
+/// const fn transmute_mut<'src, 'dst, Src, Dst>(src: &'src mut Src) -> &'dst mut Dst
+/// where
+///     'src: 'dst,
+///     Src: IntoBytes + FromBytes + NoCell,
+///     Dst: IntoBytes + FromBytes + NoCell,
```
Comment (joshlf) [inline src/lib.rs:4588]:
```suggestion
///     Src: FromBytes + IntoBytes + NoCell,
///     Dst: FromBytes + IntoBytes + NoCell,
```

Usually we list `FromBytes` before `IntoBytes`, consistent w/ the stdlib `From` and `Into`. I don't feel strongly about it, though, so feel free to leave it as-is if you prefer.

**Review by joshlf — APPROVED** [review]

**Pull request merged**